addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2988,6 +2988,19 @@ options.registry.ConditionalVisibility = options.Class.extend({
 
         this._updateCSSSelectors();
     },
+    /**
+     * Select a value or an list of values for target's data-attributes.
+     * Should be used instead of selectRecord if the visibility is not related
+     * to database values.
+     *
+     * @see this.selectClass for parameters
+     */
+    selectValues(previewMode, widgetValue, params) {
+        const values = widgetValue.split('or');
+        this.$target[0].dataset[params.saveAttribute] = JSON.stringify(values.map((value, index) => ({value: value.trim(), id: index})));
```

Comment:
As it is implemented, the select allows for only one option (either everyone or logged in) to be chosen, with everone being the default, when the M2M widget handles adding and removing multiple records. But it does not seem to be the right UI choice as indeed "Visible for Everyone", "Hidden for everyone" don't make sense...
![image](https://user-images.githubusercontent.com/58561718/131340488-cbdcc277-c33a-4bdf-9630-a4d40f959b11.png)
I can  change to using the same kind of UI as the man2one widgets, the user would be able to select if he wants the bloc visible for Logged In and/or Logged Out as if they were records

Reply:
Ask @pierre-pde for functionnal specs :wink: Indeed two options "logged in" or "logged out" makes more sense than this "Everyone" if we want to keep consistency with the rest of the options.

But what I think instead of

Logged in --- visible for / hidden for
                      logged in / logged out
                      
I would prefer

Users ---- visible for logged in / visible for logged out / visible for everyone (only one select with last choice being the default)

More logical for two records I think.

Reply:
I pushed a new version with the select that you suggested,
I'll wait for @pierre-pde 's feedback. The alternative that we discussed would be :
![image](https://user-images.githubusercontent.com/58561718/131355767-d2db1183-b855-430d-ab89-79aa1239837c.png)